tools/eslint/src/check-license.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * @license
+ * Copyright 2024 Google Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {ESLintUtils} from '@typescript-eslint/utils';
+
+const createRule = ESLintUtils.RuleCreator(name => {
+  return `https://github.com/puppeteer/puppeteer/tree/main/tools/eslint/${name}.js`;
+});
+
+const copyrightPattern = /Copyright ([0-9]{4}) Google Inc\./;
+
+const enforceLicenseRule = createRule<[], 'licenseRule'>({
+  name: 'license',
```

Comment:
Name should match filename.
```suggestion
  name: 'check-license',
```